src/datasets/search.py

```diff
@@ -384,20 +385,23 @@ def save(self, file: Union[str, PurePath]):
         else:
             index = self.faiss_index
 
-        faiss.write_index(index, str(file))
+        with fsspec.open(str(file), "wb", **(storage_options or {})) as f:
+            index = faiss.write_index(index, faiss.BufferedIOWriter(faiss.PyCallbackIOWriter(f.write)))
```

Comment:
There's no need to capture `index` here, right?

```suggestion
            faiss.write_index(index, faiss.BufferedIOWriter(faiss.PyCallbackIOWriter(f.write)))
```